internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -4451,6 +4490,45 @@ func flattenKubernetesClusterAzureMonitorProfile(input *managedclusters.ManagedC
 	}
 }
 
+func expandKubernetesClusterUpgradeSettings(input []interface{}) *managedclusters.ClusterUpgradeSettings {
+	if len(input) == 0 || input[0] == nil {
+		return nil
+	}
+
+	values := input[0].(map[string]interface{})
+
+	overrides := []managedclusters.ControlPlaneUpgradeOverride{}
+
+	for _, override := range values["control_plane_upgrade_overrides"].([]interface{}) {
+		overrides = append(overrides, managedclusters.ControlPlaneUpgradeOverride(override.(string)))
+	}
+	until := values["until"].(string)
+
+	upgradeSettings := managedclusters.ClusterUpgradeSettings{
+		OverrideSettings: &managedclusters.UpgradeOverrideSettings{
+			ControlPlaneOverrides: &overrides,
+			Until:                 &until,
+		},
+	}
+
+	return &upgradeSettings
+}
+
+func flattenKubernetesClusterUpgradeSettings(input *managedclusters.ClusterUpgradeSettings) []interface{} {
+	if input == nil || input.OverrideSettings == nil {
+		return nil
+	}
+	if input.OverrideSettings.ControlPlaneOverrides == nil || input.OverrideSettings.Until == nil {
+		return nil
+	}
+	return []interface{}{
+		map[string]interface{}{
+			"controle_plane_upgrade_overrides": *input.OverrideSettings.ControlPlaneOverrides,
```

Comment:
`controle` should probably be `control`

Reply:
This is something that is occurring multiple times - it is not really consistent whether controle or control is used.